test/new-e2e/go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/DataDog/datadog-agent/test/new-e2e
 
 go 1.22.0
 
+toolchain go1.22.3
```

Comment:
I'm not sure we use the `toolchain` directive. 

Is this something we cn remove?